landing-pages/site/content/en/blog/kubernetes-pod-operator/index.md

```diff
@@ -0,0 +1,153 @@
+- **Flexibility of configurations and dependencies** - Configuring a run-time environment to support all sets of dependencies for all tasks is a mess, that can be mitigated by only configuring run-time environments for a single task in the image description. This also means that various teams can work on different tasks without having to worry about knowing Airflow, this bringing in the much needed abstraction between the tasks and the pipeline itself.
+- **Usage of Kubernetes Secrets for added security** - Yes! Client secrets can now be safely passed on to the container when they are running.
+
+This operator can also be created in a similar manner by instantiating the [`airflow.contrib.operators.kubernetes_pod_operator.KubernetesPodOperator`](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/contrib/operators/kubernetes_pod_operator/index.html#airflow.contrib.operators.kubernetes_pod_operator.KubernetesPodOperator) with the necessary arguments.
+
+```python
+from airflow.contrib.operators.kubernetes_pod_operator import KubernetesPodOperator
+
+say_hello = KubernetesPodOperator(
+    task_id="say_hello",
+    image="registry.hub.docker.com/library/whalesay",
+    arguments=["cowsay", "hello"],
+    name="say_hello",
+    namespace="default",
+    in_cluster=False,
+        # setting image_null_policy to "Always" makes sure the container is
+        # pulled from the repositoy before every run, thus fetching the updated image
+    image_pull_policy="Always",
+        # setting node_selectors with key-value pairs will make sure that the pod
+        # is allocated to the particular cluster according to the cluster labels
+        # mentioned in the key-value pairs
+        node_selector={
+            "beta.kubernetes.io/instance-type": "t3.large",
+        },
+    is_delete_operator_pod=True, # set False if you want the pods to be persisted
+    get_logs=True,
+    dag=dag,
+)
+```
+
+# Pitfalls
+
+`KubernetesPodOperator` was actually built by Bloomberg, as a part of their continued commitment to developing the Kubernetes ecosystem, and is fairly new. It was released in mid 2018 and is still maturing.
+
+Debugging becomes difficult once a task fails because the pod is immediately deleted, so for additional logs you might need to watch for pod event updates using the Kubernetes CLI or `kubectl`.  You can also set the `is_delete_operator_pod` to `False` but that would need manual deletion of the pods later.
```

Comment:
Does this solve your problem?
See: https://github.com/apache/airflow/pull/11815

Reply:
Wow, I didn't know about this. Has it been released though?

Reply:
It will be today :)